### PR TITLE
[SMTChecker] Support events and low-level logs

### LINKED ISSUE #9785: [SMTChecker] Support events
```
Warning: Assertion checker does not yet implement this type of function call.
  --> uniswap-v2-core/contracts/UniswapV2ERC20.sol:51:14:
   |
51 |         emit Approval(owner, spender, value);
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I don't think the SMT checker needs to do anything with events though, just ignore them and don't emit a warning.

Could also include the low-levels logs, `log0()` to `log4()`.

### PR DESCRIPTION
**Pull request by axic**
Fixes #9785.

### CODE REVIEW

**Review by leonardoalt — DISMISSED** [review]
Nice!

**Comment by axic** [comment]
Actually made new tests which are more comprehensive checking for side-effects, will push them later today.

File: test/libsolidity/smtCheckerTests/special/event.sol
Comment (axic) [inline test/libsolidity/smtCheckerTests/special/event.sol:20]: Detected as failing properly.

File: test/libsolidity/smtCheckerTests/special/log.sol
Comment (axic) [inline test/libsolidity/smtCheckerTests/special/log.sol:26]: Same here.

**Review by mijovic — DISMISSED** [review]
All tests are looking good :)

**Comment by leonardoalt** [comment]
Changelog conflict, but might wanna fix only after we merge the revert PR (or make that one the target of this one)

**Review by leonardoalt — DISMISSED** [review]

**Comment by chriseth** [comment]
changelog conflict

**Comment by axic** [comment]
> changelog conflict

Because you keep merging stuff while I'm waiting for CI. And then I wait another 3 hours for CI :)

**Comment by leonardoalt** [comment]
Wat? I ain't merging nothing

**Comment by leonardoalt** [comment]
Ah, him :p

**Review by leonardoalt — DISMISSED** [review]

**Comment by axic** [comment]
Any why the cli test is running into this? Is this related to the PR?
```
SUCCESS: 880 syntaxTests passed, 0 failed, 1383 could not be compiled (2220 sources total).
Testing soljson via the fuzzer...
ASSERTION VIOLATION
File: /build/libz3-static-dev-yYOfiW/libz3-static-dev-4.8.9/src/muz/spacer/spacer_context.cpp
Line: 824
UNREACHABLE CODE WAS REACHED.
4.8.9.0
Please file an issue with this message and more detail about how you encountered it at https://github.com/Z3Prover/z3/issues/new
```

**Comment by mijovic** [comment]
> Any why the cli test is running into this? Is this related to the PR?
> 
> ```
> SUCCESS: 880 syntaxTests passed, 0 failed, 1383 could not be compiled (2220 sources total).
> Testing soljson via the fuzzer...
> ASSERTION VIOLATION
> File: /build/libz3-static-dev-yYOfiW/libz3-static-dev-4.8.9/src/muz/spacer/spacer_context.cpp
> Line: 824
> UNREACHABLE CODE WAS REACHED.
> 4.8.9.0
> Please file an issue with this message and more detail about how you encountered it at https://github.com/Z3Prover/z3/issues/new
> ```

I think this is weird z3 error. As far as I know it is not related to this PR, but @leonardoalt can confirm probably.

**Comment by leonardoalt** [comment]
Ughhhh that's annoying. That's the same thing we had in the previous version. At some point I figured the shift test from https://github.com/ethereum/solidity/pull/9807 was the culprit and removed it. In that PR I readded tests that used to be problematic, I guess one of them is causing trouble again. So annoying.

**Review by mijovic — DISMISSED** [review]

**Comment by mijovic** [comment]
I think we can merge this

**Comment by leonardoalt** [comment]
I think we need to find out the buggy test before it propagates to develop like last time.

**Comment by leonardoalt** [comment]
Rebased

**Review by leonardoalt — APPROVED** [review]

**Review by mijovic — APPROVED** [review]

**Pull request merged**
